core/docs/API.md

```diff
@@ -21,32 +21,315 @@ The API exposes functions for:
+
+Read and Write functions
+
+| Images IO                                                                                                                         | Return     |
+| --------------------------------------------------------------------------------------------------------------------------------- | ---------- |
+| `IO_Image_Read( State * , const char* file, int format=IO_Fileformat_Regular, int idx_image, int idx_chain )`                     | `void`     |
+| `IO_Image_Write( State * , const char* file, int format=IO_Fileformat_Regular, int idx_image, int idx_chain )`                    | `void`     |
+| `IO_Image_Append( State * , const char* file, int iteration, int format=IO_Fileformat_Regular, int idx_image, int idx_chain )`    | `void`     |
+
+| Chains IO                                                                                                                         | Return     |
+| --------------------------------------------------------------------------------------------------------------------------------- | ---------- |
+| `IO_Chain_Read( State * , const char* file, int idx_image, int idx_chain )`                                                       | `void`     |
+| `IO_Chain_Write( State * , const char* file, int idx_image, int idx_chain )`                                                      | `void`     |
+
+| Collection IO                                                                                                                     | Return     |
+| --------------------------------------------------------------------------------------------------------------------------------- | ---------- |
+| `IO_Collection_Read( State * , const char* file, int idx_image, int idx_chain )`                                                  | `void`     |
+| `IO_Collection_Write( State * , const char* file, int idx_image, int idx_chain )`                                                 | `void`     |
+
+Energies from `System` and `Chain`
+
+| System Energies                                                                                | Return     |
+| ---------------------------------------------------------------------------------------------- | ---------- |
+| `IO_Write_System_Energy_per_Spin( State * , const char* file, int idx_chain )`                 | `void`     |
+| `IO_Write_System_Energy( State * , const char* file, int idx_image, int idx_chain )`           | `void`     |
+
+
+| Chain Energies                                                                                 | Return     |
+| ---------------------------------------------------------------------------------------------- | ---------- |
+| `IO_Write_Chain_Energies( State * , const char* file, int idx_chain )`                         | `void`     |
+| `IO_Write_Chain_Energies_Interpolated( State * , const char* file, int idx_chain )`            | `void`     |
+
+
+### Exceptions
```

Comment:
you can remove this - the API functions should not throw exceptions (I know this is not correctly kept some times), as C does not have exceptions